SimCalorimetry/EcalSimAlgos/src/EcalShapeBase.cc

```diff
@@ -49,8 +49,11 @@ void EcalShapeBase::buildMe(const edm::EventSetup* evtSetup) {
 
   const double maxelt(1.e-5 < maxel ? maxel : 1);
 
-  for (unsigned int i(0); i != shapeArray.size(); ++i) {
-    shapeArray[i] = shapeArray[i] / maxelt;
+  if (normalize) {
+
+    for (unsigned int i(0); i != shapeArray.size(); ++i) {
```

Comment:
Range based loop.

Reply:
Done.

Reply:
Somehow I cannot get this to work with a ranged-based loop. Even your code snippet you posted as a comment to my commit reverting the range-based loop had the same behavior (it compiled and ran but the digis were nonsense.)

Reply:
OK. Maybe I overlooked something then.